packages/maskbook/src/social-network-provider/twitter.com/ui/injectPostImageRevealer.tsx

```diff
@@ -0,0 +1,68 @@
+import { Component, MouseEvent } from 'React'
+import { querySelectorAll } from '../utils/selector'
+import { renderInShadowRoot } from '../../../utils/shadow-root/renderInShadowRoot'
+import { DOMProxy } from '@dimensiondev/holoflows-kit'
+import { Flags } from '../../../utils/flags'
+
+let buttonUrls = new Set<String>()
+
+class RevealEncryptedButton extends Component<{ elem: HTMLElement }, { isHidden: boolean, buttonText: string }> {
```

Comment:
Please use function component instead of class component, thanks

Reply:
are you sure? RevealEncryptedButton has state for each url... if I convert to a functional component, wouldn't that just make state more hard to manage?

Reply:
Now function components can contain states too.

https://reactjs.org/docs/hooks-intro.html

Reply:
sick! on it.